torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -110,7 +115,7 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   // object of the ndarray to the tensor and disabling resizes on the storage.
   // This is not sufficient. For example, the tensor's storage may be changed
   // via Tensor.set_, which can free the underlying memory.
-  PyObject* py_tensor = THPVariable_Wrap(make_variable(tensor, false));
+  PyObject* py_tensor = THPVariable_Wrap(as_variable_ref(tensor).variable_data());
```

Comment:
We could add an explicit check for `tensor.is_variable()` here, but I think `as_variable_ref(tensor)` already checks whether `tensor` is a Variable.

Reply:
Do you need the `variable_data`?

Reply:
I just checked and we don't need to use `variable_data` here - resizing is not supported on either the PyTorch tensor or the numpy array, so creating a shallow-copy of the tensor being converted is not necessary in any case.